# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.h
# Commit: e030be1d2dd9
# Full Hash: e030be1d2dd9b0bb47680fac29f20447c9d66f8c
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-02-21 09:51:10
# Regressor Bug: 1616715
# File Overlap Count: 1
# Description:
#   Bug 1616715 - Make ReadPixelDesc per-call. r=lsalzman
#   
#   GetSurfaceSnapshot (and similar) want to ReadPixels the backbuffer,
#   without having to navigate the GL PIXEL_PACK state.
#   Since this is cold cold code, it's nice to reduce the backend state by
# ==============================================================================

diff -r adc841b3e204 -r e030be1d2dd9 dom/canvas/HostWebGLContext.h
--- a/dom/canvas/HostWebGLContext.h	Fri Feb 21 02:27:08 2020 +0000
+++ b/dom/canvas/HostWebGLContext.h	Fri Feb 21 03:12:41 2020 +0000
@@ -409,7 +409,7 @@
     mContext->LinkProgram(*obj);
   }
 
-  void PixelStorei(GLenum pname, GLint param) const {
+  void PixelStorei(GLenum pname, uint32_t param) const {
     mContext->PixelStorei(pname, param);
   }
 
@@ -646,15 +646,15 @@
   }
 
   // ------------------------------ Readback -------------------------------
-  void ReadPixelsPbo(GLint x, GLint y, GLsizei width, GLsizei height,
-                     GLenum format, GLenum type, uint64_t offset) const {
-    mContext->ReadPixelsPbo(x, y, width, height, format, type, offset);
+  void ReadPixelsPbo(const webgl::ReadPixelsDesc& desc,
+                     const uint64_t offset) const {
+    mContext->ReadPixelsPbo(desc, offset);
   }
 
-  void ReadPixels(GLint x, GLint y, GLsizei width, GLsizei height,
-                  GLenum format, GLenum type, RawBuffer<uint8_t>& dest) const {
+  void ReadPixels(const webgl::ReadPixelsDesc& desc,
+                  RawBuffer<uint8_t>& dest) const {
     const auto range = MakeRange(dest);
-    mContext->ReadPixels(x, y, width, height, format, type, range);
+    mContext->ReadPixels(desc, range);
   }
 
   // ----------------------------- Sampler -----------------------------------